admin/app/controllers/admin/commercial/DashboardRenderer.scala

```diff
@@ -0,0 +1,49 @@
+package controllers.admin.commercial
+
+import java.util.Locale
+
+import jobs.CommercialDfpReporting
+import jobs.CommercialDfpReporting.DfpReportRow
+import model.{ApplicationContext, NoCache}
+import play.api.mvc._
+
+object DashboardRenderer extends Results {
+
+  def renderDashboard(testName: String, dashboardTitle: String)(implicit request: RequestHeader,
+                                                                context: ApplicationContext): Result = {
+    val maybeData = for {
+      reportId                  <- CommercialDfpReporting.reportMappings.get(CommercialDfpReporting.teamKPIReport)
+      report: Seq[DfpReportRow] <- CommercialDfpReporting.getReport(reportId)
+    } yield {
+      val keyValueRows: Seq[KeyValueRevenueRow] = report.flatMap { row =>
+        val fields = row.value.split(",").toSeq
+        for {
+          customCriteria: String    <- fields.lift(0)
+          customTargetingId: String <- fields.lift(1)
+          totalImpressions: Int     <- fields.lift(2).map(_.toInt)
+          totalAverageECPM: Double  <- fields.lift(3).map(_.toDouble / 1000000.0d) // convert DFP micropounds to pounds
+        } yield KeyValueRevenueRow(customCriteria, customTargetingId, totalImpressions, totalAverageECPM)
+      }
+
+      keyValueRows
+    }
+
+    val abTestRows = maybeData.getOrElse(Seq.empty)
+
+    val controlDataRow = abTestRows.find(_.customCriteria.startsWith(s"ab=${testName}Control"))
```

Comment:
this implies the `testName` is a server side test. is that ok?

Reply:
I think it's all right as the next increment.  It sort of assumes a server-side test anyway because there's just a control and a single variant.